dev/devenv-readme.md

```diff
@@ -12,49 +12,175 @@ First off, if you're on one of the following platforms, we provide the [devenv-i
+    > ftype sh_auto_file
+    sh_auto_file="C:\Git\git-bash.exe" --no-cd "%L" %*
+    ```
+  1. If necessary, change the .sh association to use your Cygwin shell:
+
+    ```
+    # Example setting Cygwin bash to be associated with .sh files 
+    > ftype sh_auto_file="C:\cygwin64\bin\bash.exe" %1 %*
+    ```
+
+## AWS Configuration
+
+### AWS Credentials
+
+Many of the automated tests associated with the Blue Button framework use AWS resources.  Before running a build using Maven or importing projects into your Eclipse IDE, which will run a build automatically, please ensure the appropriate accounts and credentials are configured within your environment.  **This is necessary to prevent incurring unwanted charges on the wrong AWS account**.
+
+Below are links to detailed instructions on configuring your AWS credentials for your environment:
+
+  * [Configuring the AWS CLI](http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html)
+  * [Configuration and Credential Files](http://docs.aws.amazon.com/cli/latest/userguide/cli-config-files.html)
+
+## Github Configuration
+
+### Github SSH
+
+You will need to configure an SSH credential in order to clone the Blue Button repositories.  Instructions are thoroughly documented on Github but for convenience here are the relevant links:
+
+  * [Connecting to Github with SSH](https://help.github.com/articles/connecting-to-github-with-ssh/)
+  * [Generating a new SSH key and adding it to the ssh-agent](https://help.github.com/articles/generating-a-new-ssh-key-and-adding-it-to-the-ssh-agent/)
+  * [Adding a new SSH key to your GitHub account](https://help.github.com/articles/adding-a-new-ssh-key-to-your-github-account/)
+  * [Testing your SSH connection](https://help.github.com/articles/testing-your-ssh-connection/)
+
+## Maven Configuration
+
 ### Maven `toolchains.xml`
 
 Right now, the script does not create the Maven `toolchains.xml` file (though it can and should). As a workaround, create it manually yourself, as `~/.m2/toolchains.xml`, with content similar to the following (adjust the paths to match your system):
 
 ```
 <?xml version="1.0" encoding="UTF8"?>
 <toolchains>
-	<toolchain>
-		<type>jdk</type>
-		<provides>
-			<version>1.8</version>
-			<vendor>oracle</vendor>
-			<id>jdk-8u31-linux-x64</id>
-		</provides>
-		<configuration>
-			<jdkHome>/home/karl/workspaces/tools/jdk-8u31-linux-x64</jdkHome>
-		</configuration>
-	</toolchain>
+  <toolchain>
+    <type>jdk</type>
+    <provides>
+      <version>1.8</version>
+      <vendor>oracle</vendor>
+      <!-- Change the id value to match the jdk version on your system --> 
+      <id>jdk-8u31-linux-x64</id>
+    </provides>
+    <configuration>
+      <!-- Change the jdkHome value to point to the jdk location on your system --> 
+      <jdkHome>/home/myusername/workspaces/tools/jdk-8u31-linux-x64</jdkHome>
+    </configuration>
+  </toolchain>
 </toolchains>
 ```
 
+### Maven GPG Signing
```

Comment:
I'd add a reference in here to the OSSRH section below, since the two aren't next to each other anymore.

Reply:
Will add link...